### ISSUE TITLE
ggcr: https://github.com/Azure/go-autorest is deprecated

### ISSUE THREAD

**Issue by sudheermygapula** [google/go-containerregistry#2003]
### Describe the bug

https://github.com/Azure/go-autorest is deprecated , Avoid end-of-life technologies .

### Additional context

Please refere issue for more details https://github.com/argoproj/argo-workflows/issues/13506

**Comment by BenTheElder** [comment]
This was removed in kubernetes upstream: https://github.com/kubernetes/kubernetes/issues/112431

I think it's only in this repo because of k8schain.

**Comment by agilgur5** [comment]
Yes, I detailed this more in https://github.com/argoproj/argo-workflows/issues/13506#issuecomment-2310608745:

> Looks like `go-containerregistry` still has it as an indirect dep even in its `latest` version (which I tried upgrading to in the `go.mod` as well): https://github.com/google/go-containerregistry/blob/00f182b8d3070d2c0a0334657fd58a284778ef83/pkg/authn/k8schain/go.mod#L23
>
> and `docker-credential-acr-env` has a direct dep on it in its `latest`: https://github.com/chrismellard/docker-credential-acr-env/blob/82a0ddb2758901b711d9d1614755b77e401598a1/go.mod#L7

**Comment by agilgur5** [comment]
> and `docker-credential-acr-env` has a direct dep on it in its `latest`:

Seems like `docker-credential-acr-env` hasn't had a commit in almost 2 years and a release in longer: https://github.com/chrismellard/docker-credential-acr-env/issues/15

**Comment by gaganhr94** [comment]
Looks like this project `docker-credential-acr-env` is not maintained anymore. I've built https://github.com/gaganhr94/docker-credential-acr as a modern, maintained drop-in replacement for [docker-credential-acr-env](https://github.com/chrismellard/docker-credential-acr-env). It provides a good number of functionalities, and is built on the current Azure SDK (azidentity / azcontainerregistry) — eliminating the dependency on the end-of-life go-autorest stack entirely. Check it out, and let me know if anything needs to be improved in this.
